cloudify_libvirt/domain_tasks.py

```diff
@@ -97,6 +123,160 @@ def configure(**kwargs):
         conn.close()
 
 
+def _update_network_list(dom, lease_only=True):
+    if lease_only:
+        request_type = libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_LEASE
+    else:
+        request_type = libvirt.VIR_DOMAIN_INTERFACE_ADDRESSES_SRC_AGENT
+
+    # get known by libvirt interfaces
+    virt_networks = dom.interfaceAddresses(request_type)
+    ctx.logger.info("Libvirt knows about such networks: {}"
+                    .format(repr(virt_networks)))
+
+    # networks from instance
+    if 'params' not in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties['params'] = {}
+    # add networks
+    if 'networks' not in ctx.instance.runtime_properties['params']:
+        ctx.instance.runtime_properties['params']['networks'] = []
+    # instance networks
+    instance_networks = ctx.instance.runtime_properties['params']["networks"]
+
+    for virt_name in virt_networks:
+        lease = virt_networks[virt_name]
+        for vm_network in instance_networks:
+            # copy current state
+            if vm_network.get('mac') == lease.get('hwaddr'):
+                vm_network['dev'] = virt_name
+                vm_network['addrs'] = lease.get('addrs', [])
+                # force update
+                ctx.instance.runtime_properties._set_changed()
+                break
+        else:
+            if lease.get('hwaddr'):
+                instance_networks.append({
+                    'dev': virt_name,
+                    'addrs': lease.get('addrs', []),
+                    'mac': lease['hwaddr']
+                })
+                # force update
+                ctx.instance.runtime_properties._set_changed()
+
+    if ctx.instance.runtime_properties.get('ip'):
+        # we already have some ip
+        return
+
+    for vm_network in instance_networks:
+        for ip_addr in vm_network.get('addrs', []):
+            if ip_addr.get('type') == libvirt.VIR_IP_ADDR_TYPE_IPV4:
+                ctx.instance.runtime_properties['ip'] = ip_addr['addr']
+                return
+
+
+@operation
+def reboot(**kwargs):
+    ctx.logger.info("reboot")
+
+    resource_id = ctx.instance.runtime_properties.get('resource_id')
+
+    if not resource_id:
+        ctx.logger.info("No servers for reboot")
```

Comment:
@0lvin do not you think we need to raise an error here when the resource_id is not provided because at this case the operation will passed with an info log ?